src/core_plugins/kibana/public/visualize/embeddable/visualize_embeddable_factory.js

```diff
@@ -46,52 +43,57 @@ export class VisualizeEmbeddableFactory extends EmbeddableFactory {
           });
         }
 
+        let panelTitle;
         if (!container.getHidePanelTitles()) {
-          visualizeScope.sharedItemTitle = panel.title !== undefined ? panel.title : savedObject.title;
+          panelTitle = panel.title !== undefined ? panel.title : savedObject.title;
         }
-        visualizeScope.savedObj = savedObject;
-        visualizeScope.panel = panel;
 
         const parsedUiState = savedObject.uiStateJSON ? JSON.parse(savedObject.uiStateJSON) : {};
-        visualizeScope.uiState = new PersistedState({
+        const uiState = new PersistedState({
           ...parsedUiState,
           ...panel.embeddableConfig,
         });
         const uiStateChangeHandler = () => {
-          visualizeScope.panel = container.updatePanel(
-            visualizeScope.panel.panelIndex,
-            { embeddableConfig: visualizeScope.uiState.toJSON() }
+          panel = container.updatePanel(
+            panel.panelIndex,
+            { embeddableConfig: uiState.toJSON() }
           );
         };
-        visualizeScope.uiState.on('change', uiStateChangeHandler);
-
-        visualizeScope.savedObj.vis.setUiState(visualizeScope.uiState);
-
-        visualizeScope.savedObj.vis.listeners.click = this.filterBarClickHandler(container.getAppState());
-        visualizeScope.savedObj.vis.listeners.brush = this.brushEvent(container.getAppState());
-
-        // The chrome is permanently hidden in "embed mode" in which case we don't want to show the spy pane, since
-        // we deem that situation to be more public facing and want to hide more detailed information.
-        visualizeScope.getShouldShowSpyPane = () => !chrome.getIsChromePermanentlyHidden();
-
-        container.registerPanelIndexPattern(panel.panelIndex, visualizeScope.savedObj.vis.indexPattern);
-
-        const visualizationInstance = this.$compile(visualizationTemplate)(visualizeScope);
-        const rootNode = angular.element(domNode);
-        rootNode.append(visualizationInstance);
+        uiState.on('change', uiStateChangeHandler);
+
+        savedObject.vis.setUiState(uiState);
+
+        savedObject.vis.listeners.click = this.filterBarClickHandler(container.getAppState());
+        savedObject.vis.listeners.brush = this.brushEvent(container.getAppState());
+
+        container.registerPanelIndexPattern(panel.panelIndex, savedObject.vis.indexPattern);
+
+        const handler = loader.embedVisualizationWithSavedObject(domNode, savedObject, {
+          uiState: uiState,
+          // Append visualization to container instead of replacing its content
+          append: true,
+          cssClass: `panel-content ${savedObject.vis.type.name}`,
+          // The chrome is permanently hidden in "embed mode" in which case we don't want to show the spy pane, since
+          // we deem that situation to be more public facing and want to hide more detailed information.
+          showSpyPanel: !chrome.getIsChromePermanentlyHidden(),
```

Comment:
can you explain why this instead of `getShouldShowSpyPane()` which was use before ?

Reply:
`getShouldShowSpyPane` returned that value. I talked to Stacey, and there is currently no scenario where this value can actually change while a visualization is visible, so I just replaced it by setting it initially to this value. Also it wouldn't have worked beforehand properly, since changing that permanently hidden value isn't triggering any callback, and is not within the angular scope, so the `<visualize>` as rendered by dashboards would anyway not have picked up any changes (IF there would be a case where it changes, which doesn't exist).